FlexConfirmMail/Outlook
# Add support for confirming unsafe addresses

Add support to specify unsafe addresses.

If a unsafe domain entry contains "@", we regard it as address, not domain.

When an address matches an unsafe addresses rule, a warning "[警告] 注意が必要な**アドレス**（{0}）が宛先に含まれています。" is displayed as below.

![image](https://github.com/FlexConfirmMail/Outlook/assets/15982708/0ee2b339-4613-4beb-8b5d-f79d47644c7a)

For the domains case, a warning is "[警告] 注意が必要な**ドメイン**（{0}）が宛先に含まれています。" as below.

![image](https://github.com/FlexConfirmMail/Outlook/assets/15982708/4fcd84cc-03d3-4c17-8fe5-b6f1d0c7748a)

Note that warnings are displayed for each addresses, so if rules match a lot of addresses, a lot of warnings are displayed.
In case of domains, warnings are displayed per domain, so there are not so many warnings usually.

## Test

* Remove files in `%APPDATA%\FlexConfirmMail` if exist.
* Open Outlook and open setting dialog for FlexConfirmMail
* Open the "注意が必要なドメイン" tab of the setting dialog.
  * Confirm that the comment have description for how to specify addresses like below  ...OK
    ```
    # (5) ドメインだけでなくアドレス全体を指定することもできます。
    #     * @を含む場合はアドレスが指定されたとみなします。
    #     * @を含まない場合はドメインが指定されたとみなします。
    #
    ```
* Add entries as following to the "注意が必要なドメイン" setting
  ```
  test@a.co.jp
  *test@b.co.jp
  -*test@c.co.jp
  test2@*
  test.co.jp
  ```
* Try to send mail to `test@a.co.jp` and  `test3@a.co.jp` （test for `test@a.co.jp`）
  * Confirm that `test@a.co.jp` is in "添付ファイル/その他の警告" ...OK
  * Confirm that `test3@a.co.jp` is **not** in "添付ファイル/その他の警告" ...OK
* Try to send mail to `test-test@a.co.jp` and  `test-test@b.co.jp` (test for `*test@b.co.jp`)
  * Confirm that `test-test@b.co.jp` is in "添付ファイル/その他の警告" ...OK
  * Confirm that `test-test@a.co.jp` is **not** in "添付ファイル/その他の警告" ...OK
* Try to send mail to `test-test@a.co.jp` and `test-test@c.co.jp` (test for `-*test@c.co.jp`)
  * Confirm that `test-test@a.co.jp` and  `test-test@c.co.jp` are **not** in "添付ファイル/その他の警告"  ...OK
* Try to send mail to `test2@a.co.jp` and `test2@c.co.jp` (test for `test2@*`)
  * Confirm that `test2@a.co.jp` and  `test2@c.co.jp` are in "添付ファイル/その他の警告" ...OK
* Try to send mail to `test-test@test.co.jp` and `test-test@a.co.jp` (test for `test.co.jp`, domain case)
  * Confirm that **`test.co.jp`** is in "添付ファイル/その他の警告" ...OK
    *  The message shoud be "[警告] 注意が必要な**ドメイン**（test.co.jp）が宛先に含まれています。"
  * Confirm that `test-test@a.co.jp` is in **not** "添付ファイル/その他の警告" ...OK
